py-polars/polars/utils/_construction.py

```diff
@@ -1341,10 +1340,7 @@ def pandas_has_default_index(df: pd.DataFrame) -> bool:
     else:
         # finally, is the index _equivalent_ to a default unnamed
         # integer index with frame data that was previously sorted
-        return (
-            isinstance(df.index, IntegerIndex)
-            and (df.index.sort_values() == np.arange(len(df))).all()
-        )
+        return df.index.sort_values() == np.arange(len(df)).all()
```

Comment:
IntegerIndex is no more https://pandas.pydata.org/docs/dev/whatsnew/v2.0.0.html#index-can-now-hold-numpy-numeric-dtypes